### PR TITLE
Add new date fields to db and api

### PR DESCRIPTION
**Pull request by 0atman**
#### Checklist

- [x] I've used the pull request template
- [x] I've linked this PR to the relevant issue (if mission work)
- [x] I've written unit tests for these changes (if code change)
- [ ] I've updated the documentation in (If any documentation requires updating)
    - [ ] README.md
- [ ] set updated_at and created_at to the current time of the migration
- [ ] bump updated when the pages update/delete/create routes are hit

### CODE REVIEW

File: spec/db/migration_5_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  end
+  it "adds a sensible default published from v4 to v5" do
+    form_id = database[:forms].insert(name: "name", submission_email: "submission_email", org: "testorg")
+
+    migrator.migrate_to(database, 5)
+
+    updated_form = database[:forms].where(id: form_id).first
+    expect(updated_form[:published]).to eq(false)
+  end
+end
```
Comment (0atman) [inline spec/db/migration_5_spec.rb:17]: @alice-carr just looks at this. I LOVE unit testing migrations omg

File: spec/db/migration_5_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
```
Comment (0atman) [inline spec/db/migration_5_spec.rb:17]: @alice-carr just looks at this. I LOVE unit testing migrations omg

**Comment by 0atman** [comment]
@thomasiles has suggested that 'published' be a datestamp, and we update it on form updated. I don't hate this idea, but maybe it should be called "updated"? But then how do we gate publishing/draft?

**Comment by 0atman** [comment]
Maybe add:
`published_at`
and
`updated_at`
?

**Comment by DavidBiddle** [comment]
Agree that all 3 of those sound like sensible fields

**Comment by 0atman** [comment]
will do this today or tomorrow :ok_hand:

**Comment by thomasiles** [comment]
Makes sense - I guess updated at will be changed when pages are updated too? Maybe we should go with some other wording than published_at now if it's easier?

**Comment by DavidBiddle** [comment]
Feels like `made_live_at` would fit the wording we're using?

**Comment by 0atman** [comment]
`made_live_at` feels a little clunky, why not just `live_at`, especially as "made" is past tense, and we'll potentially be post-dating this?

**Comment by alice-carr** [comment]
I think `published_at` reads better than `live_at` - I can imagine finding `live_at` hard to decipher if I saw it without context

**Comment by 0atman** [comment]
@alice-carr I think the problem is that the wider team have decided on some very specific meanings for _live_ and _published_, our abstraction is leaking

**Comment by alice-carr** [comment]
@0atman yeah that makes total sense

**Comment by DavidBiddle** [comment]
> `made_live_at` feels a little clunky, why not just `live_at`, especially as "made" is past tense, and we'll potentially be post-dating this?

I agree it's a bit clunky, but `created_at`, `published_at` and `updated_at` are all past tense as well so on that front it's consistent.

Are we content designing a DB field? 😄

**Comment by 0atman** [comment]
API design counts! Created, Updated, ARE past tense, though, these events will always be set to `Time.now`, whereas `live_at` can be set for the future, right?

**Comment by 0atman** [comment]
This is now fully working, I just tested the flow.

# After initial creation
```json
{
    "id": 1,
    "name": "form1",
    "submission_email": "tris@example.com",
    "org": "an-org3",
    "created_at": "2022-08-03 15:43:46 +0100",
    "live_at": null,
    "updated_at": null,
    "start_page": null
}
```

# Editing but no live date
```json
{
    "id": 1,
    "name": "form1",
    "submission_email": "tris@example.com",
    "org": "an-org3",
    "created_at": "2022-08-03 15:43:46 +0100",
    "live_at": null,
    "updated_at": "2022-08-03 15:45:46 +0100",
    "start_page": null
}
```

# Editing and adding a live date
```json
{
    "id": 1,
    "name": "form1",
    "submission_email": "tris@example.com",
    "org": "an-org3",
    "created_at": "2022-08-03 15:43:46 +0100",
    "live_at": "2022-08-13 11:11:11 +0100",
    "updated_at": "2022-08-03 16:09:52 +0100",
    "start_page": null
}
```

@DavidBiddle @thomasiles would love your review, as you'll be consuming this - does it make sense?

**Comment by thomasiles** [comment]
Amazing! 

Two questions:
 - would it make sense to set `updated_at` and `created_at` to the current time of the migration? That's dirty but otherwise the possibility of them being empty has to be handled by clients and we don't have to worry about production (I don't think)
 - Would it make sense to bump updated when the pages update/delete/create routes are hit? Either that or add similar fields to the pages table?

**Comment by 0atman** [comment]
Aha! I'd forgotten about pages. From a user's perspective when a page is updated, the FORM is updated. Yes, i'll do that.

I see no harm with that migration suggestion - seems more coherent for current test data.

**Comment by 0atman** [comment]
Sorry I didn't check this in on Wednesday - this is ready! @thomasiles I've added tests to cover modifying a page updating the `updated_at` field as suggested :ok_hand:

File: db/migrations/4_add_org.rb [hunk 3]
```diff
@@ -1,3 +1,5 @@
+require "pry"
+
```
Comment (aliuk2012) [inline db/migrations/4_add_org.rb:2]:
```suggestion
```

File: db/migrations/5_add_published.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+require "pry"
+
```
Comment (aliuk2012) [inline db/migrations/5_add_published.rb:2]:
```suggestion
```

**Comment by aliuk2012** [comment]
Might be worth rebasing this branch with main branch (Code coverage should be enabled).

Apart from that looks really good.

File: db/migrations/4_add_org.rb (hunk 3 above)
Comment (0atman) [reply db/migrations/4_add_org.rb:2]: aha!

**Comment by thomasiles** [comment]
Looks good!

No values for existing records - I don't think that matters though as we don't have any real users :)

**Comment by 0atman** [comment]
@aliuk2012 all done plz re-review!

**Comment by aliuk2012** [comment]
@0atman I think something must of happened with the rebase because I see some of my commits in here that shouldn't relate to the changes you made. eg adding code coverage commits

**Comment by 0atman** [comment]
@aliuk2012 yeah, they're on main right?

**Comment by aliuk2012** [comment]
🤦 Sorry, I just merged my PR that had migrations that ended up causing your branch to have conflicts....more reason to move to timestamp filename model

**Comment by 0atman** [comment]
@aliuk2012 ok this looks weird but can merge cleanly I think. Sorry about the weirdness

File: db/migrations/4_add_org.rb (hunk 3 above)
Comment (aliuk2012) [reply db/migrations/4_add_org.rb:2]: hmmm still seems to have pry

**Comment by 0atman** [comment]
@aliuk2012 removed pry for real

**Pull request merged**